example-project/usage-from-c/Makefile

```diff
@@ -1,8 +1,4 @@
-# cargo-c saves all the information in the .pc file
-# Do not try to pass simply -lname since it will not work for static linking
-
-LDLIBS = `pkg-config --libs example_project`
-CFLAGS = `pkg-config --cflags example_project`
+LDLIBS = -lexample_project
```

Comment:
I fail to see how this would have anything to do with static linking ... but I'd like to find out!

How does static linking work?

I tried using `LDFLAGS = -static` and it mostly failed:

* `master` branch: https://github.com/mgeier/cargo-c/actions/runs/1679754055
* this PR: https://github.com/mgeier/cargo-c/actions/runs/1679649968 and https://github.com/mgeier/cargo-c/actions/runs/1679649973

Some of the `example-project-with-subdir` test runs passed, others didn't, but I don't know why exactly.

What am I doing wrong?
How can I get it to link statically without errors?

Reply:
`pkg-config --static --libs <package>` will provide the correct link line to use with -static. 

You can also see what happens when you pass `--library-type=staticlib` to cargo cinstall.

Reply:
Cool, I didn't know about the `--static` flag (and the `Libs.private` section)!

It doesn't seem to work on Ubuntu, though

    /usr/bin/ld: cannot find -lgcc_s

https://github.com/mgeier/cargo-c/actions/runs/1689682978
https://github.com/mgeier/cargo-c/actions/runs/1689564188
https://github.com/mgeier/cargo-c/actions/runs/1689682980

Is there probably some development package missing?

Reply:
pass `--static -Wl,-Bdynamic,-lgcc_s,-Bstatic {link-line}`, I guess libgcc_s.a is missing from ubuntu.

Reply:
Thanks!

I'm not sure how exactly this is supposed to be used.
I tried it in a few variations and always got similar errors:

<details>

```
/usr/bin/ld: attempted static link of dynamic object `/lib/x86_64-linux-gnu/libgcc_s.so.1'
/usr/bin/ld: /usr/local/lib/libexample_project.a(std-7c582493123fc1dd.std.f40690d7-cgu.0.rcgu.o): undefined reference to symbol 'pthread_cond_signal@@GLIBC_2.3.2'
/usr/bin/ld: /lib/x86_64-linux-gnu/libc.so.6: error adding symbols: DSO missing from command line
```

```
/rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c//library/std/src/sys_common/net.rs:191: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: attempted static link of dynamic object `/lib/x86_64-linux-gnu/libgcc_s.so.1'
/usr/bin/ld: /usr/local/lib/libexample_project.a(std-7c582493123fc1dd.std.f40690d7-cgu.0.rcgu.o): undefined reference to symbol '__tls_get_addr@@GLIBC_2.3'
/usr/bin/ld: /lib/x86_64-linux-gnu/ld-linux-x86-64.so.2: error adding symbols: DSO missing from command line
```

```
/rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c//library/std/src/sys/unix/os.rs:610: warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
...
/rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c//library/std/src/sys_common/net.rs:191: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: attempted static link of dynamic object `/lib/x86_64-linux-gnu/libgcc_s.so.1'
```

</details>

I've found https://stackoverflow.com/a/18803598/, which suggests a similar solution.
It also mentions the GCC flag `-fno-builtin`, but I couldn't get it to work with that either.

Do you happen to know the exact compiler invocation that works?

Reply:
@lu-zero Do you know how we can make static linking work?

Reply:
`--library-type=staticlib` produces only the `.a` so the linking works, it might be useful to fix the link-line provided by rustc to avoid this mess.